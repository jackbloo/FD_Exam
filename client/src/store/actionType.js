export const LOGIN = 'LOGIN'
export const FETCHPASSWORD = 'FETCHPASSWORD'
export const LOGOUT = 'LOGOUT'
export const FETCHDETAIL = 'FETCH DETAIL'
export const SETEDIT = 'SETEDIT'
export const SHOW = 'SHOW'
export const UNSHOW = 'UNSHOW'
export const LOAD = 'LOAD'
export const UNLOAD = 'UNLOAD'
export const DELETE = 'DELETE'