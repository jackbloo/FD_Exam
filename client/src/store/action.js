import {LOGIN, LOGOUT, FETCHPASSWORD,SHOW,UNSHOW,LOAD,UNLOAD, FETCHDETAIL} from '../store/actionType'


export const show = () => ({type:SHOW})
export const unshow = () => ({type:UNSHOW})
export const login = () => ({type:LOGIN})
export const logout = () => ({type:LOGOUT})
export const load = () => ({type:LOAD})
export const unload = () => ({type: UNLOAD})
export const fetchPassword = (data) =>  ({type:FETCHPASSWORD, data})


// export const deleteOnePass = (id) => dispatch => {
//     dispatch({type:LOAD})
//     pm
//     .doc(id)
//     .delete()
//     .then(function() {
//         dispatch({type:UNLOAD})
//         toast.success('Your Password is deleted')
//     }).catch(function(error) {
//         toast.error(error)
//     });
// }
