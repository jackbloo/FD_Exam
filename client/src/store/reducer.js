import {GETDATA} from './actionType'

const initialState = {
    editorChoice:[],
    latestArticles:[],
    latestReview:[],
    navbar: [{id:'skincare'},{id:'make up'},{id:'body'},{id:'hair'},{id:'fragrance'},{id:'nails'},{id:'tools'},{id:'brands'}],
    linkData:[{id: 'About Us'}, {id: 'Terms & Conditions'}, {id: 'Awards'},{id: 'Feedback'},{id: 'Privacy Policy'}, {id: 'Newsletter'}, {id: 'Content'},{id: 'Help'}]
}

const reducer = (state = initialState, action) => {
    switch(action.type){
        case GETDATA:
            const data = action.data
            return {
                ...state,
                editorChoice: [...data["editor's choice"]],
                latestArticles:[...data["latest articles"]],
                latestReview:[...data["latest review"]]
            }
        default:
            return state     
    }
 
}

export default reducer