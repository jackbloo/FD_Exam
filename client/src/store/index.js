import { configureStore } from '@reduxjs/toolkit'
import reducer from '../store/reducer'

const store = configureStore({reducer: reducer})

export default store