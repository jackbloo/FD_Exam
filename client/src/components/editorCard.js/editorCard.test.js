import React from 'react';
import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import EditorCard from './editorCard'
import { Provider} from 'react-redux'
import store from '../../store'


describe('Unit Testing of EditorCard', () => {
    const data = {
        editor: 'Test-1',
        published_at: '2 hours ago',
        role:'Director',
        product:{
            rating: 4.3,
            name:'Test',
            description: 'test-1'
        }
    }

    it('Should Render Card match snapshot when editor is true', ()=>{

     render(
            <Provider store={store}>
                    <EditorCard data={data} isEditor={true}/>
            </Provider>
        )
        const card = screen.getByTestId('editor-card')
        expect(card).toMatchSnapshot()
    })


    it('Should Render Card match snapshot when editor is false', ()=>{

        render(
               <Provider store={store}>
                       <EditorCard data={data} isEditor={false}/>
               </Provider>
           )
           const card = screen.getByTestId('editor-card')
           expect(card).toMatchSnapshot()
       })
   

    
    it('Should Render Card without crashing', ()=>{
     render(
            <Provider store={store}>
                    <EditorCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('editor-card')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
        expect(card).toHaveTextContent("4.3(7)Testtest-1")
        expect(card).not.toHaveTextContent('Facebook.com')
    })

    
    it('Should Render editor image', ()=>{
     render(
            <Provider store={store}>
                    <EditorCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('editor-image')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render editor title when isEditor true', ()=>{
     render(
            <Provider store={store}>
                    <EditorCard data={data} isEditor={true}/>
            </Provider>
        )
        const card = screen.getByTestId('editor-role')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render editor editor when isEditor true', ()=>{
     render(
            <Provider store={store}>
                    <EditorCard data={data} isEditor={true}/>
            </Provider>
        )
        const card = screen.getByTestId('editor-editor')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })
    
    it('Should Render editor description', ()=>{
     render(
            <Provider store={store}>
                    <EditorCard data={data}/>
            </Provider>
        )
        const card = screen.getByTestId('editor-description')
        expect(card).toBeInTheDocument()
        expect(card).toBeVisible()
    })

    it('Should Render editor name', ()=>{
        render(
               <Provider store={store}>
                       <EditorCard data={data}/>
               </Provider>
           )
           const card = screen.getByTestId('editor-name')
           expect(card).toBeInTheDocument()
           expect(card).toBeVisible()
       })


})

