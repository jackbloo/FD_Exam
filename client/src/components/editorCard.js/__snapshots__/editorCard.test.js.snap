// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing of EditorCard Should Render Card match snapshot when editor is false 1`] = `
<div
  class="sc-bdvvtL fUOFOh"
  data-testid="editor-card"
>
  <div
    class="sc-dkPtRN gcoOsN"
  >
    <img
      class="sc-furwcr bBKeGd"
      data-testid="editor-image"
      src="https://static.femaledaily.com/dyn/480/images/prod-pics/product_1525323944_Tonymoly_D_800x800.jpg"
    />
    <div
      class="sc-pVTFL IcGFG"
    >
      <div
        class="sc-jrQzAO jrErZU"
        data-testid="editor-rating"
      >
        4.3
      </div>
      <div
        class="sc-kDTinF MGmIF"
      >
        <div
          class="sc-crHmcD hbDJWj"
        >
          <i
            class="sc-iqseJM ewJeOS fa fa-star icon"
          />
          <i
            class="sc-iqseJM ewJeOS fa fa-star icon"
          />
          <i
            class="sc-iqseJM ewJeOS fa fa-star icon"
          />
          <i
            class="sc-iqseJM ewJeOS fa fa-star icon"
          />
          <i
            class="sc-iqseJM gCAcpz fa fa-star icon"
          />
        </div>
        (7)
      </div>
    </div>
    <div
      class="sc-egiyK lcmcFC"
    >
      <div
        class="sc-bqiRlB gTpuPX"
        data-testid="editor-name"
      >
        Test
      </div>
      <div
        class="sc-ksdxgE cujrOH"
        data-testid="editor-description"
      >
        test-1
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit Testing of EditorCard Should Render Card match snapshot when editor is true 1`] = `
<div
  class="sc-bdvvtL fUOFOh"
  data-testid="editor-card"
>
  <div
    class="sc-gsDKAQ chudgA"
  >
    <div
      class="sc-gKclnd fqzGzL"
    >
      <img
        class="sc-iCfMLu khgBkU"
        src="/assets/NaN.png"
      />
    </div>
    <div
      class="sc-jRQBWg bJTKsk"
    >
      <div
        class="sc-hKwDye farejx"
        data-testid="editor-editor"
      >
        Test-1
      </div>
      <div
        class="sc-eCImPb emHCaL"
        data-testid="editor-role"
      >
        Director
      </div>
    </div>
  </div>
  <div
    class="sc-dkPtRN gcoOsN"
  >
    <img
      class="sc-furwcr bBKeGd"
      data-testid="editor-image"
      src="https://static.femaledaily.com/dyn/480/images/prod-pics/product_1525323944_Tonymoly_D_800x800.jpg"
    />
    <div
      class="sc-pVTFL IcGFG"
    >
      <div
        class="sc-jrQzAO jrErZU"
        data-testid="editor-rating"
      >
        4.3
      </div>
      <div
        class="sc-kDTinF MGmIF"
      >
        <div
          class="sc-crHmcD hbDJWj"
        >
          <i
            class="sc-iqseJM ewJeOS fa fa-star icon"
          />
          <i
            class="sc-iqseJM ewJeOS fa fa-star icon"
          />
          <i
            class="sc-iqseJM ewJeOS fa fa-star icon"
          />
          <i
            class="sc-iqseJM ewJeOS fa fa-star icon"
          />
          <i
            class="sc-iqseJM gCAcpz fa fa-star icon"
          />
        </div>
        (7)
      </div>
    </div>
    <div
      class="sc-egiyK lcmcFC"
    >
      <div
        class="sc-bqiRlB gTpuPX"
        data-testid="editor-name"
      >
        Test
      </div>
      <div
        class="sc-ksdxgE cujrOH"
        data-testid="editor-description"
      >
        test-1
      </div>
    </div>
  </div>
</div>
`;
